### PR TITLE
Casper backend

### PR DESCRIPTION
**Pull request by kubaplas**

### CODE REVIEW

File: docusaurus/docs/backends/04-livenet.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+💁  INFO : Calling "hash-d26fcbd210..." with entrypoint "transfer".
+🙄  WAIT : Waiting 15s for "65b1a5d21...".
+🙄  WAIT : Waiting 15s for "65b1a5d21...".
+💁  INFO : Deploy "65b1a5d21..." successfully executed.
+Owner's balance: 4004
+Recipient's balance: 4000
+```
+Those logs are a result of the last 4 lines of the above listing.
+Each deployment or a call to the blockchain will be noted and will take some time to execute.
+We can see that the `transfer` call took over 15 seconds to execute. But calling balance_of was nearly instant
```
Comment (kpob) [inline docusaurus/docs/backends/04-livenet.md:151]:
```suggestion
We can see that the `transfer` call took over 15 seconds to execute. But calling `balance_of` was nearly instant
```

File: docusaurus/docs/backends/04-livenet.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+All calls of entrypoints executed on a Casper blockchain cost gas - even if they do not change the state.
+It is possible however to query the state of the blockchain for free.
+
+This principle is used in the Livenet backend - all calls that do not change the state of the blockchain are really executed offline - the only thing that is requested from the
+node is the current state. This is why the `balance_of` call was almost instant and free.
+
+Basically, if the entrypoint function is not mutable or does not make an external call, it is executed offline and
+node is used for the state query only.
+
+Take note, that even if the function is not mutable, but the contract is not deployed or loaded, the Livenet backend will
```
Comment (kpob) [inline docusaurus/docs/backends/04-livenet.md:169]: If I were a newcomer I could imagine what is a "deploy", but what the heck is "loading" a contract. Somewhere in the `backends` section the idea of loaders should be explained.

File: docusaurus/docs/backends/03-casper.md [hunk 3]
```diff
@@ -22,7 +22,6 @@ The other exposed types are:
 
 * `CallArgs` - wraps around casper's [`RuntimeArgs`][runtime_args];
 * `Balance` - U512 type alias;
```
Comment (kpob) [inline docusaurus/docs/backends/03-casper.md:24]: No `Balance` in the new version

File: docusaurus/docs/backends/03-casper.md [hunk 4]
```diff
@@ -22,7 +22,6 @@ The other exposed types are:
 
 * `CallArgs` - wraps around casper's [`RuntimeArgs`][runtime_args];
```
Comment (kpob) [inline docusaurus/docs/backends/03-casper.md:23]: No `CallArgs` in the new version, we've got `CallDef`, but it is not just a wrapper, holds a more fields

File: docusaurus/docs/backends/03-casper.md [hunk 5]
```diff
@@ -41,21 +40,9 @@ Under the hood, Odra integrates with [Casper Event Standard] and creates a few [
    
 Besides that, all the events the contract emits are registered - events schemas are written to the storage under the `__events_schema` key.
 
-:::note
-Don't forget to expose events in the module using `#[odra::module(events = [...])]`.
```
Comment (kpob) [inline docusaurus/docs/backends/03-casper.md:45]: You have to list the events to register them in a contract, I wouldn't remove that.

File: docusaurus/docs/backends/04-livenet.md (hunk 2 above)
Comment (kubaplas) [reply docusaurus/docs/backends/04-livenet.md:169]: I've rewritten this section and added another one to cross calls doc.

File: docusaurus/docs/backends/03-casper.md (hunk 3 above)
Comment (kubaplas) [reply docusaurus/docs/backends/03-casper.md]: Removed this section altogether.

**Review by kpob — APPROVED** [review]

File: docusaurus/docs/basics/10-cross-calls.md [hunk 6]
```diff
@@ -83,6 +83,18 @@ Analogously to modules, Odra creates the `AdderRef` struct (but do not create th
...
 
+### Loading the contract
+Sometimes it is useful to load the deployed contract instead of deploying it by ourselves. This is especially useful when we want to test
+our contracts in [Livenet](../backends/04-livenet.md) backend. We can load the contract using `load` method on the `Deployer`:
+
+```rust title="examples/bin/erc20_on_livenet.rs"
+fn _load(env: &HostEnv) -> Erc20HostRef {
+    let address = "hash-d26fcbd2106e37be975d2045c580334a6d7b9d0a241c2358a4db970dfd516945";
+    let address = Address::from_str(address).unwrap();
+    Erc20Deployer::load(env, address)
```
Comment (kpob) [inline docusaurus/docs/basics/10-cross-calls.md:94]:
```suggestion
    <Erc20HostRef as HostRefLoader>::load(env, address)
```

File: docusaurus/docs/backends/04-livenet.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+    // The main difference between other backends - we need to specify
+    // the gas limit for each action.
+    // The limit will be used for every consecutive action
+    // until we change it.
+    env.set_gas(100_000_000_000u64);
+    
+    // Deploy the contract. The API is the same as in the OdraVM backend.
+    let mut token = Erc20Deployer::init(
+        env, name, symbol, decimals, Some(initial_supply)
+    );
```
Comment (kpob) [inline docusaurus/docs/backends/04-livenet.md:94]:
```suggestion
    let init_args = Erc20InitArgs {
        name,
        symbol,
        decimals,
        initial_supply: Some(initial_supply)
    };
    let mut token = Erc20HostRef::deploy(env, init_args);
```

**Pull request merged**